src/components/memberships/MembershipsPage.js

```diff
@@ -10,34 +10,66 @@ class MembershipsPage extends React.Component {
   componentDidMount() {
     this.props.loadMemberships(this.props.user_email, this.props.f2f_date)
   }
+
+  copyEmails = () => {
+    this.input.select();
+    document.execCommand('copy');
+    this.input.blur();
+  }
+
+  cardHeight = (elements) => {
+    const levelsCount = Math.ceil(elements.length / 3);
+    const firstLevelHeight = 284;
+    const nextLevelHeight = 102;
+    if (levelsCount < 1 ) {
+      return firstLevelHeight;
+    }else{
+      return firstLevelHeight + (levelsCount-1) * nextLevelHeight
+    }
+  }
+
   render() {
     const groups = {};
-    this.props.memberships.forEach(membership => {
+    const { memberships } = this.props;
+    memberships.forEach(membership => {
       groups[membership.user_role] = groups[membership.user_role] || []
       groups[membership.user_role].push(membership)
     });
-
-    const cardHeight = (elements) => {
-      const levelsCount = Math.ceil(elements.length / 3);
-      const firstLevelHeight = 282;
-      const nextLevelHeight = 102;
-      if (levelsCount < 1 ) {
-        return firstLevelHeight;
-      }else{
-        return firstLevelHeight + (levelsCount-1) * nextLevelHeight
-      }
-    }
-
+    const emails = memberships.map(m => m.user_email).join(', ')
     const { quartInOut } = easings;
     const { enter, entered, exit } = enterExitStyle.fromLeftToRight;
 
+    const grid = () => {
+      return <CSSGrid
+        component="ul"
+        columns={3}
+        columnWidth={374}
+        gutterWidth={5}
+        gutterHeight={5}
+        layout={layout.pinterest}
+        duration={800}
+        easing={quartInOut}
+        enter={enter}
+        entered={entered}
+        exit={exit}
+      >
+        {Object.keys(groups).map(roleName => {
+          return (
+            <li key={roleName} itemHeight={this.cardHeight(groups[roleName])}>
+              <MembershipList memberships={groups[roleName]} roleName={roleName} />
+            </li>
+          )
+        })}
+      </CSSGrid>
+    }
+
```

Comment:
Put the whole grid rendering thing outside the render
```
renderGrid = () => (
 (...)
)
```
and use it like `this.renderGrid()`

I don't see a point in storing it in a variable inside the render method.